src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -1,81 +1,65 @@
 package edu.kis.powp.jobs2d.command.manager;
 
-import java.util.Iterator;
 import java.util.List;
 
-import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
 import edu.kis.powp.jobs2d.command.DriverCommand;
-import edu.kis.powp.jobs2d.command.ICompoundCommand;
 import edu.kis.powp.observer.Publisher;
 
 /**
  * Driver command Manager.
  */
 public class DriverCommandManager {
-	private DriverCommand currentCommand = null;
 
-	private Publisher changePublisher = new Publisher();
+    private DriverCommand currentCommand = null;
 
-	/**
-	 * Set current command.
-	 * 
-	 * @param commandList Set the command as current.
-	 */
-	public synchronized void setCurrentCommand(DriverCommand commandList) {
-		this.currentCommand = commandList;
-		changePublisher.notifyObservers();
-	}
+    private Publisher changePublisher = new Publisher();
 
-	/**
-	 * Set current command.
-	 * 
-	 * @param commandList list of commands representing a compound command.
-	 * @param name        name of the command.
-	 */
-	public synchronized void setCurrentCommand(List<DriverCommand> commandList, String name) {
-		setCurrentCommand(new ICompoundCommand() {
+    /**
+     * Set current command.
+     *
+     * @param commandList
+     *            Set the command as current.
+     */
+    public synchronized void setCurrentCommand(DriverCommand commandList) {
+        this.currentCommand = commandList;
+        changePublisher.notifyObservers();
+    }
 
-			List<DriverCommand> driverCommands = commandList;
+    /**
+     * Set current command.
+     *
+     * @param commandList
+     *            list of commands representing a compound command.
+     * @param name
+     *            name of the command.
+     */
+    public synchronized void setCurrentCommand(List<DriverCommand> commandList, String name) {
+        setCurrentCommand(new CompoundCommand());
```

Comment:
what happens with commandList and name ?! They are lost :(

Reply:
bug fix